imap/jmap_mailbox.c

```diff
@@ -752,37 +752,48 @@ struct jmap_mailbox_get_cb_rock {
     strarray_t *sublist;
 };
 
-static int jmap_mailbox_get_cb(const mbentry_t *mbentry, void *_rock)
+static int is_jmap_mailbox(const mbentry_t *mbentry, int tombstones)
 {
-    struct jmap_mailbox_get_cb_rock *rock = _rock;
-    jmap_req_t *req = rock->req;
-    json_t *list = (json_t *) rock->get->list, *obj;
-
     /* Don't list special-purpose mailboxes. */
     if (mbtypes_unavailable(mbentry->mbtype) ||
-        mbtype_isa(mbentry->mbtype) != MBTYPE_EMAIL) {
+        mbtype_isa(mbentry->mbtype) != MBTYPE_EMAIL)
         return 0;
-    }
+
+    if ((mbentry->mbtype & MBTYPE_DELETED) && !tombstones)
```

Comment:
This doesn't do anything, because:

```
#define mbtypes_unavailable(mbtype) \
  ((mbtype) & (MBTYPE_REMOTE | MBTYPE_RESERVE | MBTYPE_MOVING | MBTYPE_DELETED))
```

We already returned on the previous check.

Reply:
Thanks! I added a commit that removes `DELETED` from the `unavailable` check in mboxlist.h. We only used it at two locations, once in jmap_mail and once here in jmap_mailbox.